tests/v2/validation/charts/backup_restore/backup_restore.go

```diff
@@ -0,0 +1,321 @@
+	userList := []*management.User{}
+	projList := []*management.Project{}
+	roleList := []*management.RoleTemplate{}
+
+	for i := 0; i < resourceCount; i++ {
+		u, err := users.CreateUserWithRole(client, users.UserConfig(), globalUserRole)
+		if err != nil {
+			return userList, projList, roleList, err
+		}
+		userList = append(userList, u)
+
+		p, err := createProject(client, clusterID)
+		if err != nil {
+			return userList, projList, roleList, err
+		}
+		projList = append(projList, p)
+
+		rt, err := createRole(client, clusterRoleContext, roleName, rules)
+		if err != nil {
+			return userList, projList, roleList, err
+		}
+		roleList = append(roleList, rt)
+	}
+
+	return userList, projList, roleList, nil
+}
+
+func createBackup(client *rancher.Client, name string) (*v1.SteveAPIObject, error) {
+	backup := setBackupObject()
+	backupTemplate := bv1.NewBackup("", name, backup)
+	completedBackup, err := client.Steve.SteveType(backupSteveType).Create(backupTemplate)
+
+	return completedBackup, err
+}
+
+func createRestore(client *rancher.Client, fileName string) (*v1.SteveAPIObject, error) {
+	restore := setRestoreObject()
+	restoreTemplate := bv1.NewRestore("", "", restore)
+	restoreTemplate.Spec.BackupFilename = fileName
+	completedRestore, err := client.Steve.SteveType(restoreSteveType).Create(restoreTemplate)
+
+	return completedRestore, err
+}
+
+func validateAWSS3BackupExists(bucket string, folder string, key string) (bool, error) {
+	if len(folder) > 0 {
+		folder += key
+		isPresent, err := checkAWSS3Object(bucket, folder)
+		return isPresent, err
+	}
+	isPresent, err := checkAWSS3Object(bucket, key)
+	return isPresent, err
+}
+
+func checkAWSS3Object(bucket string, key string) (bool, error) {
+	sdkConfig, err := awsConfig.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return false, err
+	}
+
+	s3Client := s3.NewFromConfig(sdkConfig)
+	_, err = s3Client.HeadObject(context.TODO(), &s3.HeadObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+	})
+
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case "NotFound":
+				return false, nil
+			default:
+				return false, err
+			}
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func deleteAWSBackup(bucket string, folder string, key string) error {
+	sdkConfig, err := awsConfig.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return err
+	}
+	svc := s3.NewFromConfig(sdkConfig)
+
+	input := &s3.DeleteObjectInput{}
+
+	input = &s3.DeleteObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+	}
+	if len(folder) > 0 {
+		folder += key
+		input = &s3.DeleteObjectInput{
+			Bucket: aws.String(bucket),
+			Key:    aws.String(folder),
+		}
+	}
+
+	_, err = svc.DeleteObject(context.TODO(), input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			default:
+				logrus.Error(aerr.Error())
+			}
+		} else {
+			logrus.Error(err.Error())
```

Comment:
why the switch statement here? also does this even need the if/else statement, why not just log the normal error?